src/kakarot/precompiles/precompiles.cairo

```diff
@@ -110,6 +101,19 @@ namespace Precompiles {
         );
         jmp call_precompile;
 
+        pre_kakarot_precompile:
+        let is_cairo_contract_call = Helpers.is_zero(
+            FIRST_KAKAROT_PRECOMPILE_ADDRESS - evm_address
+        );
+        let is_whitelisted = KakarotPrecompiles.is_caller_whitelisted(caller_address);
+        tempvar is_authorized = is_cairo_contract_call * is_whitelisted;
+        tempvar syscall_ptr = syscall_ptr;
+        tempvar pedersen_ptr = pedersen_ptr;
+        tempvar range_check_ptr = range_check_ptr;
+        jmp kakarot_precompile if is_authorized != 0;
+        jmp unauthorized_call if is_cairo_contract_call != 0;
+        jmp kakarot_precompile;
```

Comment:
useless as it's just the next instruction

Reply:
I prefer to keep it explicit and avoid unwanted next-instruction fallbacks